### PR TITLE
Fix travis 

### PR DESCRIPTION
**Pull request by DanielGuramulta**
Updated dependecies for travis and also added a docker build for ubuntu 18.04 (bionic).

Signed-off-by: Daniel Guramulta <daniel.guramulta@analog.com>

### CODE REVIEW

File: CI/travis/before_install_darwin.sh [hunk 1]
```diff
@@ -18,28 +18,28 @@ brew_install_or_upgrade() {
 }
 
 PYTHON="python3 python@2 python brew-pip"
-PACKAGES="qt@5.11.2 cmake fftw bison gettext autoconf automake libtool libzip glib libusb $PYTHON"
+PACKAGES="qt cmake fftw bison gettext autoconf automake libtool libzip glib libusb $PYTHON"
```
Comment (commodo) [inline CI/travis/before_install_darwin.sh]: `qt@5.11.2` i wouldn't try to force this, as things change between distros

File: CI/travis/package_darwin.sh [hunk 2]
```diff
@@ -44,7 +44,7 @@ sudo install_name_tool -change ${libusbpath} @executable_path/../Frameworks/${li
 
 sudo /usr/local/opt/qt/bin/macdeployqt Scopy.app
 
-brew install python@2
+brew install python@2 || 1
```
Comment (commodo) [inline CI/travis/package_darwin.sh]: this could hide an error;

**Comment by DanielGuramulta** [comment]
Changelog V1 -> V2:

- Check the version of python2 in case installation fails (common reason: already installed )
- Travis will now get the "common" scripts from libiio
- Added Centos 7 docker build

File: CI/travis/before_install_darwin.sh [hunk 3]
```diff
@@ -73,7 +83,7 @@ qmake_build_wget "qwtpolar-1.1.1" "https://downloads.sourceforge.net/project/qwt
 
 cmake_build_wget "volk-1.3" "http://libvolk.org/releases/volk-1.3.tar.gz"
 
-cmake_build_git "gnuradio" "https://github.com/analogdevicesinc/gnuradio" "signal_source_phase_rebased" "-DENABLE_INTERNAL_VOLK:BOOL=OFF -DENABLE_GR_FEC:BOOL=OFF -DENABLE_GR_DIGITAL:BOOL=OFF -DENABLE_GR_DTV:BOOL=OFF -DENABLE_GR_ATSC:BOOL=OFF -DENABLE_GR_AUDIO:BOOL=OFF -DENABLE_GR_CHANNELS:BOOL=OFF -DENABLE_GR_NOAA:BOOL=OFF -DENABLE_GR_PAGER:BOOL=OFF -DENABLE_GR_TRELLIS:BOOL=OFF -DENABLE_GR_VOCODER:BOOL=OFF"
+cmake_build_git "gnuradio" "https://github.com/analogdevicesinc/gnuradio" "scopy" "-DENABLE_INTERNAL_VOLK:BOOL=OFF -DENABLE_GR_FEC:BOOL=OFF -DENABLE_GR_DIGITAL:BOOL=OFF -DENABLE_GR_DTV:BOOL=OFF -DENABLE_GR_ATSC:BOOL=OFF -DENABLE_GR_AUDIO:BOOL=OFF -DENABLE_GR_CHANNELS:BOOL=OFF -DENABLE_GR_NOAA:BOOL=OFF -DENABLE_GR_PAGER:BOOL=OFF -DENABLE_GR_TRELLIS:BOOL=OFF -DENABLE_GR_VOCODER:BOOL=OFF"
```
Comment (AlexandraTrifan) [inline CI/travis/before_install_darwin.sh]: should the branch for gnuradio be "signal_source_phase_rebase", so it's consistent in all the builds?

File: CI/travis/before_install_linux.sh [hunk 4]
```diff
@@ -83,17 +113,113 @@ qmake_build_git "qwt" "https://github.com/osakared/qwt.git" "qwt-6.1-multiaxes"
 
 qmake_build_wget "qwtpolar-1.1.1" "https://downloads.sourceforge.net/project/qwtpolar/qwtpolar/1.1.1/qwtpolar-1.1.1.tar.bz2" "qwtpolar.pro" "patch_qwtpolar_linux"
 
-cmake_build_git "gnuradio" "https://github.com/analogdevicesinc/gnuradio" "signal_source_phase_rebased" "-DENABLE_INTERNAL_VOLK:BOOL=OFF -DENABLE_GR_FEC:BOOL=OFF -DENABLE_GR_DIGITAL:BOOL=OFF -DENABLE_GR_DTV:BOOL=OFF -DENABLE_GR_ATSC:BOOL=OFF -DENABLE_GR_AUDIO:BOOL=OFF -DENABLE_GR_CHANNELS:BOOL=OFF -DENABLE_GR_NOAA:BOOL=OFF -DENABLE_GR_PAGER:BOOL=OFF -DENABLE_GR_TRELLIS:BOOL=OFF -DENABLE_GR_VOCODER:BOOL=OFF"
+cmake_build_git "gnuradio" "https://github.com/analogdevicesinc/gnuradio" "scopy" "-DENABLE_INTERNAL_VOLK:BOOL=OFF -DENABLE_GR_FEC:BOOL=OFF -DENABLE_GR_DIGITAL:BOOL=OFF -DENABLE_GR_DTV:BOOL=OFF -DENABLE_GR_ATSC:BOOL=OFF -DENABLE_GR_AUDIO:BOOL=OFF -DENABLE_GR_CHANNELS:BOOL=OFF -DENABLE_GR_NOAA:BOOL=OFF -DENABLE_GR_PAGER:BOOL=OFF -DENABLE_GR_TRELLIS:BOOL=OFF -DENABLE_GR_VOCODER:BOOL=OFF"
```
Comment (AlexandraTrifan) [inline CI/travis/before_install_linux.sh]: the same issue as mentioned before, with the branch?

Comment (DanielGuramulta) [reply CI/travis/before_install_linux.sh]: For now we will be using the "scopy" branch on gnuradio like appveyor does. There are no major differences that would make the change worth.

Comment (AlexandraTrifan) [reply CI/travis/before_install_linux.sh]: Ok. But we need to change it in the centos build as well. This way we don't use different branches.

File: CI/travis/before_install_linux.sh [hunk 5]
```diff
@@ -83,17 +113,113 @@ qmake_build_git "qwt" "https://github.com/osakared/qwt.git" "qwt-6.1-multiaxes"
...
+	make_build_wget "libsigrokdecode-0.4.1" "http://sigrok.org/download/source/libsigrokdecode/libsigrokdecode-0.4.1.tar.gz"
+
+
+	qmake_build_git "qwt" "https://github.com/osakared/qwt.git" "qwt-6.1-multiaxes" "qwt.pro" "patch_qwt"
+
+	qmake_build_wget "qwtpolar-1.1.1" "https://downloads.sourceforge.net/project/qwtpolar/qwtpolar/1.1.1/qwtpolar-1.1.1.tar.bz2" "qwtpolar.pro" "patch_qwtpolar_linux"
+
+	cmake_build_git "gnuradio" \
+	"https://github.com/analogdevicesinc/gnuradio" \
+	"signal_source_phase_rebased" \
```
Comment (AlexandraTrifan) [inline CI/travis/before_install_linux.sh]: We should have "scopy" here, right?

Comment (DanielGuramulta) [reply CI/travis/before_install_linux.sh]: Yes. I will make the change.

**Comment by DanielGuramulta** [comment]
Changelog V2 -> V3:

- centos 7 build uses the "scopy" branch of gnuradio

**Pull request merged**